code/modules/chemistry/tools/robospray.dm

```diff
@@ -28,16 +29,19 @@
 	attack_self(mob/user as mob)
 		picker = picker % length(botreagents) + 1
 		currentreagent = botreagents[picker]
-		user.show_text("[src] is now injecting [currentreagent], [botreagents[currentreagent]] units left.", "blue")
+		var/datum/reagent/temp_reagent = reagents_cache[currentreagent]
+		propername = temp_reagent.name
+		user.show_text("[src] is now injecting [propername], [botreagents[currentreagent]] units left.", "blue")
+		return
+
+	get_desc(dist)
+		. += "It is injecting [propername]."
```

Comment:
Should probably be able to see volume of current reagent

Reply:
next commit